File: citaria_backend/src/main/java/com/citaria/model/Usuario.java
package com.citaria.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import java.time.LocalDateTime;
import java.util.Objects;

/**
 * Persona registrada en la aplicación.
 */
@Entity
@Table(name = "usuario")
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "organizacion_id", nullable = false)
    private Organizacion organizacion;

    @Column(nullable = false, length = 255)
    private String email;

    @Column(name = "password_hash", nullable = false, length = 255)
    private String passwordHash;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private RolUsuario rol;

    @Column(nullable = false)
    private Boolean activo = true;

    @Column(name = "email_verificado", nullable = false)
    private Boolean emailVerificado = false;

    @Column(name = "ultimo_acceso")
    private LocalDateTime ultimoAcceso;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cliente_id", unique = true)
    private Cliente cliente;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "empleado_id", unique = true)
    private Empleado empleado;

    public Integer getId() { return id; }
    public void setId(Integer id) { this.id = id; }

    public Organizacion getOrganizacion() { return organizacion; }
    public void setOrganizacion(Organizacion organizacion) { this.organizacion = organizacion; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public String getPasswordHash() { return passwordHash; }
    public void setPasswordHash(String passwordHash) { this.passwordHash = passwordHash; }

    public RolUsuario getRol() { return rol; }
    public void setRol(RolUsuario rol) { this.rol = rol; }

    public Boolean getActivo() { return activo; }
    public void setActivo(Boolean activo) { this.activo = activo; }

    public Boolean getEmailVerificado() { return emailVerificado; }
    public void setEmailVerificado(Boolean emailVerificado) { this.emailVerificado = emailVerificado; }

    public LocalDateTime getUltimoAcceso() { return ultimoAcceso; }
    public void setUltimoAcceso(LocalDateTime ultimoAcceso) { this.ultimoAcceso = ultimoAcceso; }

    public Cliente getCliente() { return cliente; }
    public void setCliente(Cliente cliente) { this.cliente = cliente; }

    public Empleado getEmpleado() { return empleado; }
    public void setEmpleado(Empleado empleado) { this.empleado = empleado; }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Usuario usuario)) return false;
        return Objects.equals(id, usuario.id) && Objects.equals(organizacion, usuario.organizacion) && Objects.equals(email, usuario.email) && Objects.equals(passwordHash, usuario.passwordHash) && rol == usuario.rol && Objects.equals(activo, usuario.activo) && Objects.equals(emailVerificado, usuario.emailVerificado) && Objects.equals(ultimoAcceso, usuario.ultimoAcceso) && Objects.equals(cliente, usuario.cliente) && Objects.equals(empleado, usuario.empleado);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, organizacion, email, passwordHash, rol, activo, emailVerificado, ultimoAcceso, cliente, empleado);
    }

    @Override
    public String toString() {
        return "Usuario{" +
                "id=" + id +
                ", organizacionId=" + (organizacion != null ? organizacion.getId() : null) +
                ", email='" + email + '\'' +
                ", rol=" + rol +
                ", activo=" + activo +
                '}';
    }
}

File: citaria_frontend/lib/data/models/usuario.dart
import 'package:citaria_frontend/data/enums/rol_usuario.dart';

class Usuario {
  final int? id;
  final int? organizacionId;
  final String email;
  final RolUsuario rol;
  final bool? activo;
  final bool? emailVerificado;
  final DateTime? ultimoAcceso;
  final int? clienteId;
  final int? empleadoId;

  const Usuario({
    this.id,
    this.organizacionId,
    required this.email,
    required this.rol,
    this.activo,
    this.emailVerificado,
    this.ultimoAcceso,
    this.clienteId,
    this.empleadoId,
  });

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      id: json['id'] == null ? null : (json['id'] as num).toInt(),
      organizacionId: json['organizacionId'] == null
          ? null
          : (json['organizacionId'] as num).toInt(),
      email: json['email'] as String,
      rol: RolUsuario.fromJson(json['rol'] as String),
      activo: json['activo'] as bool?,
      emailVerificado: json['emailVerificado'] as bool?,
      ultimoAcceso: json['ultimoAcceso'] == null
          ? null
          : DateTime.parse(json['ultimoAcceso'] as String),
      clienteId: json['clienteId'] == null
          ? null
          : (json['clienteId'] as num).toInt(),
      empleadoId: json['empleadoId'] == null
          ? null
          : (json['empleadoId'] as num).toInt(),
    );
  }
}
